sql/src/main/java/io/crate/expression/scalar/arithmetic/IntervalTimestampArithmeticScalar.java

```diff
@@ -23,29 +23,82 @@
 package io.crate.expression.scalar.arithmetic;
 
 import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
 import io.crate.metadata.FunctionIdent;
 import io.crate.metadata.FunctionInfo;
 import io.crate.metadata.Scalar;
 import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
 import io.crate.types.DataType;
+import io.crate.types.DataTypes;
 import io.crate.types.IntervalType;
 import org.joda.time.DateTime;
 import org.joda.time.DateTimeZone;
 import org.joda.time.Period;
 
-import java.util.Arrays;
+import java.util.List;
 import java.util.function.BiFunction;
 
-public class IntervalTimestampScalar extends Scalar<Long, Object> implements BiFunction<Long, Period, Long> {
+public class IntervalTimestampArithmeticScalar extends Scalar<Long, Object> implements BiFunction<Long, Period, Long> {
+
+    public static void register(ScalarFunctionModule module) {
+        for (var timestampType : List.of(DataTypes.TIMESTAMP, DataTypes.TIMESTAMPZ)) {
+            module.register(
+                Signature.scalar(
+                    ArithmeticFunctions.Names.ADD,
+                    DataTypes.INTERVAL.getTypeSignature(),
+                    timestampType.getTypeSignature(),
+                    timestampType.getTypeSignature()
+                ).withForbiddenCoercion(),
+                args -> new IntervalTimestampArithmeticScalar(
+                    "+",
+                    ArithmeticFunctions.Names.ADD,
+                    args,
+                    args.get(1))
+            );
+            module.register(
+                Signature.scalar(
+                    ArithmeticFunctions.Names.ADD,
+                    timestampType.getTypeSignature(),
+                    DataTypes.INTERVAL.getTypeSignature(),
+                    DataTypes.INTERVAL.getTypeSignature()
```

Comment:
Timestamp + interval should return a timestamp,  no?

Reply:
yes, that's right. it wasn't failing coz the return type of signature is not used. instead, we use the first arg data type like https://github.com/crate/crate/pull/9864/commits/b6348cee15c4d1290b256763a3b07c689c418da5#diff-97b9d1e99e7fb4ad3e7374a71e373f83L70